src/app/clusters/operational-credentials-server/operational-credentials-server.cpp

```diff
@@ -432,53 +455,51 @@ namespace {
 // TODO: Manage ephemeral RCAC/ICAC/NOC storage to avoid a full FabricInfo being needed here.
 FabricInfo gFabricBeingCommissioned;
 
-CHIP_ERROR SendNOCResponse(app::CommandHandler * commandObj, EmberAfNodeOperationalCertStatus status, uint8_t index,
-                           CharSpan debug_text)
+CHIP_ERROR SendNOCResponse(app::CommandHandler * commandObj, const ConcreteCommandPath & path, OperationalCertStatus status,
+                           uint8_t index, const CharSpan & debug_text)
 {
-    app::ConcreteCommandPath path = { emberAfCurrentEndpoint(), OperationalCredentials::Id, Commands::NOCResponse::Id };
-    TLV::TLVWriter * writer       = nullptr;
-
-    VerifyOrReturnError(commandObj != nullptr, CHIP_ERROR_INCORRECT_STATE);
-
-    ReturnErrorOnFailure(commandObj->PrepareCommand(path));
-    writer = commandObj->GetCommandDataIBTLVWriter();
-    ReturnErrorOnFailure(writer->Put(TLV::ContextTag(0), status));
-    if (status == EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_SUCCESS)
+    Commands::NOCResponse::Type payload;
+    payload.statusCode = status;
+    if (status == OperationalCertStatus::kSuccess)
     {
-        ReturnErrorOnFailure(writer->Put(TLV::ContextTag(1), index));
+        payload.fabricIndex.Emplace(index);
     }
-    ReturnErrorOnFailure(writer->PutString(TLV::ContextTag(2), debug_text));
-    return commandObj->FinishCommand();
+    if (!debug_text.empty())
+    {
+        payload.debugText.Emplace(debug_text);
+    }
+
+    return commandObj->AddResponseData(path, payload);
 }
 
-EmberAfNodeOperationalCertStatus ConvertToNOCResponseStatus(CHIP_ERROR err)
+OperationalCertStatus ConvertToNOCResponseStatus(CHIP_ERROR err)
 {
     if (err == CHIP_NO_ERROR)
     {
-        return EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_SUCCESS;
+        return OperationalCertStatus::kSuccess;
     }
     else if (err == CHIP_ERROR_INVALID_PUBLIC_KEY)
     {
-        return EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_INVALID_PUBLIC_KEY;
+        return OperationalCertStatus::kInvalidPublicKey;
     }
     else if (err == CHIP_ERROR_INVALID_FABRIC_ID || err == CHIP_ERROR_WRONG_NODE_ID)
     {
-        return EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_INVALID_NODE_OP_ID;
+        return OperationalCertStatus::kInvalidNodeOpId;
     }
     else if (err == CHIP_ERROR_CA_CERT_NOT_FOUND || err == CHIP_ERROR_CERT_PATH_LEN_CONSTRAINT_EXCEEDED ||
              err == CHIP_ERROR_CERT_PATH_TOO_LONG || err == CHIP_ERROR_CERT_USAGE_NOT_ALLOWED || err == CHIP_ERROR_CERT_EXPIRED ||
              err == CHIP_ERROR_CERT_NOT_VALID_YET || err == CHIP_ERROR_UNSUPPORTED_CERT_FORMAT ||
              err == CHIP_ERROR_UNSUPPORTED_ELLIPTIC_CURVE || err == CHIP_ERROR_CERT_LOAD_FAILED ||
              err == CHIP_ERROR_CERT_NOT_TRUSTED || err == CHIP_ERROR_WRONG_CERT_SUBJECT)
     {
-        return EMBER_ZCL_NODE_OPERATIONAL_CERT_STATUS_INVALID_NOC;
+        return OperationalCertStatus::kInvalidNOC;
```

Comment:
This method should generate a `DebugText` as well, that contains the original error code ID, so we don't lose the more specific error

Reply:
Filed https://github.com/project-chip/connectedhomeip/issues/14707 to do as a followup.